_examples/ai/main.go

```diff
@@ -0,0 +1,169 @@
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to init AI instance: %v", err)
+			return
+		}
+
+		aiJsonResultStr, err := aiCaller.Run("@cf/meta/llama-3.1-8b-instruct", map[string]any{
+			"prompt": "What is the origin of the phrase Hello, World",
+		})
+
+		if err != nil {
+			fmt.Println(w, "failed to get result from AI\n", err)
+			return
+		}
+
+		fmt.Println(aiJsonResultStr)
+
+		io.Copy(w, strings.NewReader(aiJsonResultStr))
+	})
+
+	http.HandleFunc("/ai-text-to-image", func(w http.ResponseWriter, req *http.Request) {
+
+		// initialize AI namespace instance
+		aiCaller, err := ai.New("AI")
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to init AI instance: %v", err)
+			os.Exit(1)
+		}
+
+		aiJsonResultStr, err := aiCaller.Run("@cf/black-forest-labs/flux-1-schnell", map[string]any{
+			"prompt": "a cyberpunk lizard",
+		})
+
+		if err != nil {
+			fmt.Println(w, "failed to get result from AI\n", err)
+			return
+		}
+
+		fmt.Println(aiJsonResultStr)
+
+		var response struct {
+			Image string `json:"image"`
+		}
+		err = json.Unmarshal([]byte(aiJsonResultStr), &response)
+
+		// Decode the base64 string
+		imageData, err := base64.StdEncoding.DecodeString(response.Image)
+		if err != nil {
+			http.Error(w, "failed to decode image", http.StatusInternalServerError)
+			return
+		}
+
+		// Set the appropriate content type for the image
+		w.Header().Set("Content-Type", "image/png")
+
+		// Write the image data to the response
+		w.Write(imageData)
+	})
+
+	http.HandleFunc("/ai-image-to-image", func(w http.ResponseWriter, req *http.Request) {
+
+		// initialize AI namespace instance
+		aiCaller, err := ai.New("AI")
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to init AI instance: %v", err)
+			os.Exit(1)
+		}
+
+		r, err := fetch.NewRequest(req.Context(), "GET", "https://pub-1fb693cb11cc46b2b2f656f51e015a2c.r2.dev/dog.png", nil)
+		if err != nil {
+			handleError(w, http.StatusInternalServerError, "Internal Error")
+			log.Printf("failed to initialize proxy request: %v\n", err)
+			return
+		}
+
+		cli := fetch.NewClient()
+		resp, err := cli.Do(r, nil)
+		if err != nil {
+			handleError(w, http.StatusInternalServerError, "Internal Error")
+			log.Printf("failed to execute proxy request: %v\n", err)
+			return
+		}
+
+		defer resp.Body.Close()
+
+		imgBytes, err := io.ReadAll(resp.Body)
+		if err != nil {
+			http.Error(w, "Error reading image", http.StatusInternalServerError)
+			return
+		}
+
+		aiResult, err := aiCaller.RunReader("@cf/runwayml/stable-diffusion-v1-5-img2img", map[string]any{
+			"prompt": "Change to a lion",
+			// "image":  imgBytes,
+			"image_b64": base64.StdEncoding.EncodeToString(imgBytes),
+		})
+
+		io.Copy(w, aiResult)
+
+		// aiResultStr, err := io.ReadAll(aiResult)
+		// if err != nil {
+		// 	http.Error(w, "Error reading AI result", http.StatusInternalServerError)
+		// 	return
+		// }
+		// reader := strings.NewReader(string(aiResultStr))
+
+		// // Leer todo como string
```

Comment:
Is this commented code a valuable snippet? If so, can you write the comments in English?

Reply:
Hi! (as comented in discord) i leave this part as "incomplete" example of how to decode the return from this ai worker from cloudfare ... waiting to @syumai to help to discover how to treat "this sreaming response" to transleate a real image.... take a look at this part ... https://github.com/syumai/workers/pull/174/files/2ea1211dc55274f32cd979b860c753aa9cfcc8bf#r2054969323 where i comment the result and want help for how to decode this response to a real image...

Reply:
update the comments for wait help :)